### PR TITLE
mlir/LogicalResult: move into llvm

### PR DESCRIPTION
**Pull request by artagnon**
This patch is part of a project to move the Presburger library into LLVM.

### CODE REVIEW

**Review by joker-eph — APPROVED** [review]
Can we keep the MLIR header and just define the `using llvm::LogicalResult;` there in the meantime? It'll avoid breaking the world downstream for now.

**Comment by artagnon** [comment]
Hi @joker-eph, could you kindly check that the stop-gap stub is what you meant?

**Review by joker-eph — APPROVED** [review]
LG!
Probably worth getting another approval, maybe @clementval can look at how it looks for customers from the Flang side?

**Review by matthias-springer — APPROVED** [review]

File: mlir/include/mlir/Dialect/IRDL/IRDLLoading.h [hunk 1]
```diff
@@ -13,16 +13,20 @@
 #ifndef MLIR_DIALECT_IRDL_IRDLREGISTRATION_H
 #define MLIR_DIALECT_IRDL_IRDLREGISTRATION_H
 
-namespace mlir {
+#include "llvm/Support/LogicalResult.h"
+namespace llvm {
 struct LogicalResult;
+} // namespace llvm
```
Comment (Moxinilian) [inline mlir/include/mlir/Dialect/IRDL/IRDLLoading.h:19]: If the file is included here, is the forward declaration still necessary?

**Review by Moxinilian — COMMENTED** [review]
Looks good to me with respects to IRDL, Mem2Reg and SROA, beyond the nit below.

**Pull request merged**